# Ouvrir les fichiers CSV avec Pandas

In [1]:
import pandas as pd
import numpy as np
#pd.set_option("max_rows", None)  #Préciser le nombre de lignes max

In [2]:
df_population = pd.read_csv("data/fr_population.csv")
df_animaux = pd.read_csv("data/fr_animaux.csv")
df_vegetaux = pd.read_csv("data/fr_vegetaux.csv")
df_cereales = pd.read_csv("data/fr_céréales.csv")
df_sous_alimentation = pd.read_csv("data/fr_sousalimentation.csv")

## Aperçu génaral des données

In [3]:
# df_population.head(5)

In [4]:
# df_population.describe()

In [5]:
# df_animaux.head(5)

In [6]:
# df_animaux.describe()

In [7]:
# df_vegetaux.head(5)

In [8]:
# df_vegetaux.describe()

In [9]:
# df_cereales.head()

In [10]:
# df_cereales.describe()

In [11]:
# df_sous_alimentation.describe()

# Réponse 1  : Nombre total d’humains sur la planète

In [12]:
df_population2013 = df_population[df_population['Code année']==2013].copy()
df_population2013['Valeur'].sum() * 1000
#L'unité utilisé dans notre db est 1000 personnes

8413993000

<ul>
    <li> Pour le nombre total d'humains en 2013, nous avons obtenue : <b>8 413 993 000</b> personnes </li>
    <li>Or, selon <a href = https://www.worldbank.org/>la banque mondiale</a> la population mondiale en 2013 est environ 7 Md d'humaines </li>
    <li>Afin de clarifier notre résultat nous allons :      
        <ol>            
            <li>Vérifier si le nombre de population pour un pays donnée est calculé plusieurs fois</li>
            <li>Les valeurs manquantes pour année ou code année</li>
            <li>Observer les données <a href = https://www.fao.org/faostat/fr/#definitions>non officielles</a>  </li>                </ol>
    </li> 
</ul>

In [13]:
df_population2013['Zone'].value_counts()

Serbie                       1
Belize                       1
Fédération de Russie         1
Zambie                       1
République centrafricaine    1
                            ..
Espagne                      1
Bulgarie                     1
Kenya                        1
Azerbaïdjan                  1
Maurice                      1
Name: Zone, Length: 175, dtype: int64

In [14]:
df_population2013['Code année'].isna().sum(),df_population2013['Année'].isna().sum()

(0, 0)

In [15]:
df_population2013[~df_population2013['Symbole'].isna()]

Code Domaine                                            Domaine  Code zone  \
33         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        351   

     Zone  Code Élément            Élément  Code Produit     Produit  \
33  Chine           511  Population totale          2501  Population   

    Code année  Année           Unité   Valeur Symbole  \
33        2013   2013  1000 personnes  1416667       A   

                               Description du Symbole  
33  Agrégat, peut inclure des données officielles,...

In [16]:
df_Chine_A = df_population[df_population['Zone'].str.match(pat = '(^Chin.)')] 
df_Chine_A

Code Domaine                                            Domaine  Code zone  \
33         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        351   
34         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...         96   
35         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        128   
36         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...         41   
37         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        214   

                         Zone  Code Élément            Élément  Code Produit  \
33                      Chine           511  Population totale          2501   
34   Chine - RAS de Hong-Kong           511  Population totale          2501   
35       Chine - RAS de Macao           511  Population totale          2501   
36        Chine, continentale           511  Population totale          2501   
37  Chine, Taiwan Province de           511  Population totale          2501   

       Produit  Code année  Année           Unité   Valeur Symbole  \
33  Population        2013   2013  1000 personnes  1416667       A   
34  Population        2013   2013  1000 personnes     7204     NaN   
35  Population        2013   2013  1000 personnes      566     NaN   
36  Population        2013   2013  1000 personnes  1385567     NaN   
37  Population        2013   2013  1000 personnes    23330     NaN   

                               Description du Symbole  
33  Agrégat, peut inclure des données officielles,...  
34                                  Donnée officielle  
35                                  Donnée officielle  
36                                  Donnée officielle  
37                                  Donnée officielle

In [17]:
df_Chine_A[df_Chine_A['Zone']!='Chine']['Valeur'].sum()

1416667

Nous remarquons que la population la zone chine est une agrégation de plusieurs autres zones, si nous calculons la somme de population de ces dernières, nous retouvons qu'elle est égale à la population chinoise, c'est pourquoi nous allons calculer la population mondiale sans prendre en considération la chine

<b> Suppression des aggrégations de la chine de toutes les dataframes </b>


In [18]:
df_population2013_sans_agg = df_population2013[~df_population2013['Code zone'].isin([41, 96,128,214])].copy()
df_animaux = df_animaux[~df_animaux['Code zone'].isin([41, 96,128,214])].copy()
df_vegetaux = df_vegetaux[~df_vegetaux['Code zone'].isin([41, 96,128,214])].copy()
df_cereales = df_cereales[~df_cereales['Code zone'].isin([41, 96,128,214])].copy()
df_sous_alimentation = df_sous_alimentation[~df_sous_alimentation['Code zone'].isin([41, 96,128,214])].copy()

In [19]:
#df_population2013_sans_chine = df_population2013[df_population2013['Zone']!='Chine']
# Pour confirmer que nous avons bien supprimer la chine de notre db:
#df_population2013_sans_agg[df_population2013_sans_agg['Zone'].str.match(pat = '(^Chin.)')]

In [20]:
# nb_population = df_population2013_sans_agg['Valeur'].sum() * 1000
nb_population = df_population2013_sans_agg['Valeur'].sum() * 1000
print(nb_population)

6997326000


=========================================================================== <br>
<B><I>Le nombre total d'humain en 2013 est 6997326000 </I></B> <br> ===========================================================================

<b> Suppression de la chine des autres bases de données </b>

In [21]:
# df_animaux = df_animaux[df_animaux['Zone']!='Chine'].copy()
# df_vegetaux = df_vegetaux[df_vegetaux['Zone']!='Chine'].copy()
# df_cereales = df_cereales[df_cereales['Zone']!='Chine'].copy()
# df_sous_alimentation = df_sous_alimentation[df_sous_alimentation['Zone']!='Chine'].copy()


# Réponse 2 : Equation à 3 termes

=========================================================================== <br>
<b><i>Disponibilité intérieure = <br>Production + Importations + |Variation de stock| - Exportations = 
<br>Aliments pour animaux + Semences + Pertes + Traitement + Autres utilisations (non alimentaire) + Nourriture
 </i></b>
=========================================================================== <br>


<strong><u> Exemple blé en france 2018 </u></strong>

In [22]:
# Données récupérées à partir du lien : https://www.fao.org/faostat/fr/#data/FBS
data_ble_france2018= pd.read_csv('data/ble_france_2018.csv')

In [23]:
data_ble_france2018.head(3)

Code Domaine                      Domaine  Code zone (FAO)    Zone  \
0          FBS  Bilans Alimentaires (2014-)               68  France   
1          FBS  Bilans Alimentaires (2014-)               68  France   
2          FBS  Bilans Alimentaires (2014-)               68  France   

   Code Élément                  Élément  Code Produit          Produit  \
0          5511               Production          2511  Blé et produits   
1          5611  Importations - Quantité          2511  Blé et produits   
2          5072       Variation de stock          2511  Blé et produits   

   Code année  Année               Unité  Valeur Symbole  \
0        2018   2018  Milliers de tonnes   35798      Im   
1        2018   2018  Milliers de tonnes    2349      Im   
2        2018   2018  Milliers de tonnes   -1819      Im   

                              Description du Symbole  
0  Données de la FAO basées sur une méthodologie ...  
1  Données de la FAO basées sur une méthodologie ...  
2  Données de la FAO basées sur une méthodologie ...

In [24]:
data_ble_france2018 = data_ble_france2018[['Code Élément', 'Élément', 'Unité', 'Valeur']].copy()

In [25]:
#Le premier terme
terme1=data_ble_france2018[data_ble_france2018['Code Élément']==5301]['Valeur'].iloc[0] #valeur de disponibilité intérieur

In [26]:
#Le second terme
terme2=data_ble_france2018[data_ble_france2018['Code Élément']==5511]['Valeur'].iloc[0] + \
data_ble_france2018[data_ble_france2018['Code Élément']==5611]['Valeur'].iloc[0] + \
data_ble_france2018[data_ble_france2018['Code Élément']==5072]['Valeur'].abs() .iloc[0]- \
data_ble_france2018[data_ble_france2018['Code Élément']==5911]['Valeur'].iloc[0]

In [27]:
#Le troisième terme
terme3=data_ble_france2018[data_ble_france2018['Code Élément']==5521]['Valeur'].iloc[0] +\
    data_ble_france2018[data_ble_france2018['Code Élément']==5527]['Valeur'].iloc[0] + \
    data_ble_france2018[data_ble_france2018['Code Élément']==5123]['Valeur'].iloc[0]+ \
    data_ble_france2018[data_ble_france2018['Code Élément']==5131]['Valeur'].iloc[0] + \
    data_ble_france2018[data_ble_france2018['Code Élément']==5154]['Valeur'].iloc[0]+ \
    data_ble_france2018[data_ble_france2018['Code Élément']==5142]['Valeur'].iloc[0]

In [28]:
print('Terme1 = ',terme1)
print('Terme2 = ',terme2)
print('Terme2 = ',terme3)

Terme1 =  19521
Terme2 =  19520
Terme2 =  19521


 <b>Exemple blé en france en 2013 (à partir de notre db): voir projet3 archivé </b>

# Réponse 3 : Calculer pour chaque pays et chaque produit la disponibilité alimentaire en kcal puis en kg de protéines.

## Etape 1  : garder uniquement les colonnes importantes et rajouter une colonne "type" pour distinguer entre df_animaux et df_vegetaux

In [29]:
df_population1 = df_population2013_sans_agg[['Code zone','Valeur','Unité']].copy()

In [30]:
#df_animaux1 = df_animaux [['Code zone', 'Zone','Produit', 'Disponibilité alimentaire (Kcal/personne/jour)', 'Disponibilité de protéines en quantité (g/personne/jour)']]
df_animaux1 = df_animaux.assign(Type = 'animal')

In [31]:
#df_vegetaux1 = df_vegetaux[['Code zone', 'Zone', 'Produit', 'Disponibilité alimentaire (Kcal/personne/jour)', 'Disponibilité de protéines en quantité (g/personne/jour)']]
df_vegetaux1 = df_vegetaux.assign(Type = 'vegetal')

## Etape 2  : concaténation entre df_animaux1 et df_vegetaux1

In [32]:
df_Alimentation1=pd.concat([df_animaux1,df_vegetaux1])

In [33]:
print('df_animaux1.shape =\t', df_animaux1.shape , '\ndf_vegetaux2.shape =\t',
      df_vegetaux1.shape,'\ndf_Alimentation1.shape =\t',df_Alimentation1.shape)

df_animaux1.shape =	 (3643, 19) 
df_vegetaux2.shape =	 (11682, 19) 
df_Alimentation1.shape =	 (15325, 19)


## Etape3 : Jointure à gauche entre df_Alimentation1, df_population1 

In [34]:
df_prinicpale = df_Alimentation1.merge( df_population1, on='Code zone', how="left", suffixes=('_produit','_population'))
print("df_Alimentation1.shape =\t",df_Alimentation1.shape,'\ndf_population1.shape =\t',df_population1.shape,
      '\ndf_alimentation_population.shape =',df_prinicpale.shape)

df_Alimentation1.shape =	 (15325, 19) 
df_population1.shape =	 (171, 3) 
df_alimentation_population.shape = (15325, 21)


In [35]:
df_prinicpale.rename(columns={'Valeur' : 'Population'}, inplace=True)
df_prinicpale.head(3)

Zone  Code zone           Produit  Aliments pour animaux (kg/an)  \
0  Afghanistan          2  Abats Comestible                            NaN   
1  Afghanistan          2      Beurre, Ghee                            NaN   
2  Afghanistan          2             Crème                            NaN   

   Autres utilisations (non alimentaire) (kg/an)  \
0                                            NaN   
1                                            NaN   
2                                            NaN   

   Disponibilité alimentaire (Kcal/personne/jour)  \
0                                             5.0   
1                                            23.0   
2                                             3.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.17        
2                                               0.60        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               2.61               
2                                               0.32               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.03          
2                                               0.04          

   Disponibilité intérieure (kg/an)  ...  Importations - Quantité (kg/an)  \
0                        53000000.0  ...                              NaN   
1                        36000000.0  ...                              0.0   
2                        18000000.0  ...                       18000000.0   

   Nourriture (kg/an)  Pertes (kg/an)  Production (kg/an)  Semences (kg/an)  \
0          53000000.0             NaN          53000000.0               NaN   
1          36000000.0             NaN          36000000.0               NaN   
2          18000000.0             NaN                 NaN               NaN   

   Traitement (kg/an)  Variation de stock (kg/an)    Type Population  \
0                 NaN                         NaN  animal      30552   
1                 NaN                         NaN  animal      30552   
2                 NaN                         NaN  animal      30552   

            Unité  
0  1000 personnes  
1  1000 personnes  
2  1000 personnes  

[3 rows x 21 columns]

## Etape 4 : Calcul de disponibilité alimentaire en kcal puis en kg 

### 1. Disponibilité alimentaire en kcal 
<ul>
    <li>Disponibilite alimenatire en kcal = <br>'Disponibilité alimentaire (Kcal/personne/jour)' * Nombre d'habitants * unité (1000) * 365 jours</li>
</ul>

In [36]:
## df_prinicpale['Nb_Kcal'] = \
df_prinicpale['Disponibilite alimentaire (kcal/an)'] = df_prinicpale['Disponibilité alimentaire (Kcal/personne/jour)'] * 365 * df_prinicpale['Population'] * 1000

### 2. Disponibilité alimentaire en kg de protéine 
<ul>
    <li>Disponibilite alimenatire en kg de proteine = <br>Disponibilité de protéines en quantité (g/personne/jour) / 1000 (g --> kg) * Nombre d'habitants * unité (1000) * 365 jours</li>
</ul>
Nb : Pour optimiser le calcul, nous n'allons pas multiplier par 1000 puis diviser par 1000

In [37]:
# df_prinicpale['Nb_Kg_proteine'] = \
df_prinicpale['Disponibilite alimentaire proteine (kg/an)'] = df_prinicpale['Disponibilité de protéines en quantité (g/personne/jour)'] * 365 * df_prinicpale['Population'] 

In [38]:
df_prinicpale[['Zone','Produit','Disponibilite alimentaire (kcal/an)','Disponibilite alimentaire proteine (kg/an)']]

Zone                 Produit  \
0       Afghanistan        Abats Comestible   
1       Afghanistan            Beurre, Ghee   
2       Afghanistan                   Crème   
3       Afghanistan  Graisses Animales Crue   
4       Afghanistan      Lait - Excl Beurre   
...             ...                     ...   
15320  Îles Salomon                  Sésame   
15321  Îles Salomon                     Thé   
15322  Îles Salomon                 Tomates   
15323  Îles Salomon                     Vin   
15324  Îles Salomon          Épices, Autres   

       Disponibilite alimentaire (kcal/an)  \
0                             5.575740e+10   
1                             2.564840e+11   
2                             3.345444e+10   
3                             1.338178e+11   
4                             1.115148e+12   
...                                    ...   
15320                         0.000000e+00   
15321                         0.000000e+00   
15322                         0.000000e+00   
15323                         0.000000e+00   
15324                         8.190600e+08   

       Disponibilite alimentaire proteine (kg/an)  
0                                      8586639.60  
1                                       334544.40  
2                                       446059.20  
3                                       111514.80  
4                                     68247057.60  
...                                           ...  
15320                                         NaN  
15321                                     8190.60  
15322                                         NaN  
15323                                         NaN  
15324                                    30714.75  

[15325 rows x 4 columns]

# Réponse 4 : 

## 1. Ratio "énergie/poids" (en kcal/kg) pour chaque produit et chaque pays

In [39]:
df_ratio = df_prinicpale[['Zone','Produit','Nourriture (kg/an)','Disponibilite alimentaire (kcal/an)','Disponibilite alimentaire proteine (kg/an)']].copy()

<b> a . Remplacer les valeurs égales à 0 par des valeurs manquantes pour éviter la division sur 0 </b> 

In [40]:
df_ratio.replace(0, np.nan, inplace=True)

In [41]:
df_ratio['Ratio Energie/poids (kcal/kg)'] = df_ratio['Disponibilite alimentaire (kcal/an)']/  df_ratio["Nourriture (kg/an)"]

## 2. Ratio "poids de protéines/poids total" (en g/kg) pour chaque produit et chaque pays

In [42]:
df_ratio['Ratio protéine/poids'] = df_ratio['Disponibilite alimentaire proteine (kg/an)']/ df_ratio["Nourriture (kg/an)"]

In [43]:
df_ratio = df_ratio.loc[:,['Produit','Ratio Energie/poids (kcal/kg)','Ratio protéine/poids']].copy()

<Strong>Exemple :</Strong> <br>
Dans <a href='https://fr.wikipedia.org/wiki/%C5%92uf_(aliment)'> Wikipedia </a> on trouve que l'apport énergitique <b> des oeufs crus</b> est de 147 kcal pour 100g, soit 1470 par kg

In [44]:
df_ratio[df_ratio['Produit']=='Oeufs']

Produit  Ratio Energie/poids (kcal/kg)  Ratio protéine/poids
5      Oeufs                    1393.935000              0.105939
22     Oeufs                    1419.396632              0.114566
44     Oeufs                    1331.866750              0.105681
65     Oeufs                    1222.768481              0.104162
87     Oeufs                    1405.130381              0.114204
...      ...                            ...                   ...
3545   Oeufs                    1373.376494              0.106769
3567   Oeufs                    1322.116905              0.098931
3589   Oeufs                    1401.723257              0.106881
3610   Oeufs                    1010.201912              0.111122
3632   Oeufs                    1023.825000              0.081906

[171 rows x 3 columns]

<Strong>Exemple :</Strong> <br>
Dans <a href ='https://fr.wikipedia.org/wiki/Avoine_cultiv%C3%A9e'> Wikipedia </a> on trouve que<b> l'avoine</b> contient 10,70 g de proteine pour 100g, soit un ratio de 0,107

In [45]:
df_ratio[df_ratio['Produit']=='Avoine']

Produit  Ratio Energie/poids (kcal/kg)  Ratio protéine/poids
3697   Avoine                    2407.905000              0.077053
3770   Avoine                            NaN                   NaN
3840   Avoine                            NaN                   NaN
3912   Avoine                    3090.863110              0.114124
3984   Avoine                            NaN                   NaN
...       ...                            ...                   ...
14915  Avoine                            NaN                   NaN
14987  Avoine                            NaN                   NaN
15057  Avoine                    2322.192128              0.095332
15130  Avoine                    1754.587893              0.072954
15203  Avoine                    2410.306316              0.114490

[164 rows x 3 columns]

# Réponse 5 :

<b> a .Moyenne du ratio pour chaque aliment </b> 

In [46]:
df_ratio = df_ratio.groupby('Produit').mean().reset_index()

<b> Les 5 aliments les plus calorique </b>

In [47]:
Elts_caloriques = df_ratio.sort_values(by='Ratio Energie/poids (kcal/kg)', ascending=False).head(5)
Elts_caloriques[['Produit','Ratio Energie/poids (kcal/kg)']]

Produit  Ratio Energie/poids (kcal/kg)
45  Huiles de Foie de Poisso                    9982.932500
42       Huile de Son de Riz                    9379.568137
43           Huile de Sésame                    9314.546692
46        Huiles de Poissons                    9003.715504
34          Huile d'Arachide                    8925.582634

<b>Les 5 aliments les plus protéinés</b>

In [48]:
Elts_proteinees = df_ratio.sort_values(by='Ratio protéine/poids', ascending=False).head(5)
Elts_proteinees[['Produit','Ratio protéine/poids']]


Produit  Ratio protéine/poids
27  Graines Colza/Moutarde              0.285365
81                    Soja              0.285156
6   Arachides Decortiquees              0.251573
70                    Pois              0.223219
50     Légumineuses Autres              0.220334

<b>Exemples précedents (Selon la moyenne):</b>

In [49]:
df_ratio[df_ratio['Produit']=='Oeufs']

Produit  Ratio Energie/poids (kcal/kg)  Ratio protéine/poids
58   Oeufs                    1343.440349              0.105473

In [50]:
df_ratio[df_ratio['Produit']=='Avoine']

Produit  Ratio Energie/poids (kcal/kg)  Ratio protéine/poids
7  Avoine                     2135.35579              0.084294

<b> Rajout dans la df principale les ratios </b>

In [51]:
df_prinicpale = df_prinicpale.merge(df_ratio, on='Produit').copy()

#  Réponse 6 : Disponibilité intérieure mondiale pour les produits végétaux 

In [52]:
df_alimentation_vegetal = df_prinicpale[df_prinicpale['Type'] == 'vegetal'].copy()

In [53]:
#df_vegetal_interieur = df_alimentation_vegetal[['Zone','Produit','Disponibilité intérieure (kg/an)','Ratio Energie/poids (kcal/kg)','Ratio protéine/poids']].copy()

<b> a. Disponibilité intérieure mondiale en kcal (pour les produits végétaux uniquement)</b>

In [54]:
#df_alimentation_vegetal.columns

In [55]:
df_alimentation_vegetal['Disponibilité intérieure (kcal/an)'] = df_alimentation_vegetal["Disponibilité intérieure (kg/an)"] * \
    df_alimentation_vegetal["Ratio Energie/poids (kcal/kg)"] 
total_disponibilite_interieur_vegetal_kcal = df_alimentation_vegetal['Disponibilité intérieure (kcal/an)'].sum()
print('Disponibilité intérieure mondiale en kcal pour les produits végétaux uniquement = ',total_disponibilite_interieur_vegetal_kcal)

Disponibilité intérieure mondiale en kcal pour les produits végétaux uniquement =  1.3780136677764184e+16


============================================================================================================ <br>
Disponibilité intérieure mondiale en kcal pour les produits végétaux uniquement =  <b>1.3780136677764184e+16</b> <br>
============================================================================================================

<b> a. Disponibilité intérieure mondiale en proteine (pour les produits végétaux uniquement)

In [56]:
df_alimentation_vegetal['Disponibilité intérieure en proteine (g/an)'] = df_alimentation_vegetal["Disponibilité intérieure (kg/an)"] * \
    df_alimentation_vegetal["Ratio protéine/poids"] * 1000
total_disponibilite_interieur_vegetal_proteine = df_alimentation_vegetal['Disponibilité intérieure en proteine (g/an)'].sum()
print('total_disponibilite_interieur_vegetal_proteine = ',total_disponibilite_interieur_vegetal_proteine)

total_disponibilite_interieur_vegetal_proteine =  349177827738287.5


============================================================================================================ <br>
Disponibilité intérieure mondiale en proteine pour les produits végétaux uniquement =  <b>349177827738287.5 g</b> <br>
============================================================================================================

# Réponse 7 : Nombre d'humains qui pourraient être nourris si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture 

<b>a. Selon le nombre de calories :</b><br>

In [57]:
nombre_personne_nourris_disponibilite_interieur_vegetal_kcal =total_disponibilite_interieur_vegetal_kcal / (2500 * 365)
pourcentage_nourris_disponibilite_interieur_vegetal_kcal = nombre_personne_nourris_disponibilite_interieur_vegetal_kcal *100/nb_population
print ('Nombre de personnes nourries en utilisant la disponibilite interieur vegetal selon de nombre de calories =', nombre_personne_nourris_disponibilite_interieur_vegetal_kcal)
print('Pourcentage des personnes nourries en utilisant la disponibilite interieur vegetal selon de nombre de calories =', pourcentage_nourris_disponibilite_interieur_vegetal_kcal,'%')

Nombre de personnes nourries en utilisant la disponibilite interieur vegetal selon de nombre de calories = 15101519646.864859
Pourcentage des personnes nourries en utilisant la disponibilite interieur vegetal selon de nombre de calories = 215.81843759837483 %


Sachant que la ration alimentaire moyenne nécessaire est de <b>2 500 kcal/personne/jour</b>. [Source <a href="https://fr.wikipedia.org/wiki/Ration_alimentaire#cite_note-2">Wikipedia</a> ] , alors :<br>
Le nombre de personnes nourries en utilisant la disponibilite interieur vegetal selon de nombre de calories = <b>15101519646</b><br>
Pourcentage des personnes nourries en utilisant la disponibilite interieur vegetal selon de nombre de calories = <b>215.89 %</b>

<b>Selon le nombre de proteines : </b>

In [58]:
nombre_personne_nourris_disponibilite_interieur_vegetal_proteine =total_disponibilite_interieur_vegetal_proteine / (49 * 365)
pourcentage_nourris_disponibilite_interieur_vegetal_proteine = nombre_personne_nourris_disponibilite_interieur_vegetal_proteine *100/nb_population
print ('nombre_personne_nourris_disponibilite_interieur_vegetal_proteine =', nombre_personne_nourris_disponibilite_interieur_vegetal_proteine)
print('pourcentage_nourris_disponibilite_interieur_vegetal_proteine =', pourcentage_nourris_disponibilite_interieur_vegetal_proteine,'%')

nombre_personne_nourris_disponibilite_interieur_vegetal_proteine = 19523501690.706596
pourcentage_nourris_disponibilite_interieur_vegetal_proteine = 279.0137502626946 %


Les besoins moyens en protéines ont été définis par la FAO qui recommande <b>49 g</b> de protéines pour l'hommes adultes [<a href = https://fr.wikipedia.org/wiki/Prot%C3%A9ine>Source</a>].
nombre_personne_nourris_disponibilite_interieur_vegetal_proteine = 19523501690.706596 <br>
pourcentage_nourris_disponibilite_interieur_vegetal_proteine = 279.0137502626946 %

In [59]:
#  L'ANSES recommande un apport nutritionnel conseillé (ANC) de 0,83 g·kg-1·j-1, pour un maximum de 2,2 g·kg-1·j-1 chez l’adulte en bonne santé, soit 62 g par jour pour un homme de 75 kg. Il faut noter que les ANC sont supérieurs aux besoins moyens qui sont de 0,66 g·kg-1·j-1 selon ce même rapport, ce qui donnerait 49,5 g par jour pour le cas précédent.


# Question 8 : Combien d'humains pourraient être nourris si toute la disponibilité alimentaire en produits végétaux, la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture ? Donnez les résultats en termes de calories, puis de protéines, et exprimez ensuite ces 2 résultats en pourcentage de la population mondiale.

<b> a. Selon le nombre de calories</b> :

In [60]:
perte_animaux_kcal = (df_alimentation_vegetal.loc[:,'Aliments pour animaux (kg/an)']+df_alimentation_vegetal.loc[:,'Pertes (kg/an)']) \
                 * df_alimentation_vegetal.loc[:,"Ratio Energie/poids (kcal/kg)"] 
df_alimentation_vegetal.loc[:,'Disponibilite sans perte (kcal)'] = perte_animaux_kcal + df_alimentation_vegetal.loc[:,'Disponibilite alimentaire (kcal/an)'] 
total_disponibilite_sans_perte_vegetal_kcal = df_alimentation_vegetal['Disponibilite sans perte (kcal)'].sum()
print("total_disponibilite_sans_perte_vegetal_kcal = ",total_disponibilite_sans_perte_vegetal_kcal)

total_disponibilite_sans_perte_vegetal_kcal =  6334924922323592.0


total_disponibilite_sans_perte_vegetal_kcal =  6334924922323592.0

In [61]:
#df_disponibilité_sans_perte_kcal = df_alimentation_vegetal[['Zone','Code zone','Produit','Disponibilite sans perte']]

In [62]:
nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal =total_disponibilite_sans_perte_vegetal_kcal / (2500 * 365)
pourcentage_nourris_disponibilite_sans_perte_vegetal_kcal = nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal *100/nb_population
print ('nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal =', nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal)
print('pourcentage_nourris_disponibilite_sans_perte_vegetal_kcal =', pourcentage_nourris_disponibilite_sans_perte_vegetal_kcal,'%')

nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal = 6942383476.519005
pourcentage_nourris_disponibilite_sans_perte_vegetal_kcal = 99.21480686363627 %


nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal = 6942383476.519005<br>
pourcentage_nourris_disponibilite_sans_perte_vegetal_kcal = 99.21480686363627 %

<b> a. Selon le nombre de proteines</b> :

In [63]:
perte_animaux_proteine = (df_alimentation_vegetal.loc[:,'Aliments pour animaux (kg/an)']+df_alimentation_vegetal.loc[:,'Pertes (kg/an)']) \
                 * df_alimentation_vegetal.loc[:,"Ratio protéine/poids"] 
df_alimentation_vegetal.loc[:,'Disponibilite sans perte proteine(kg/an)'] = perte_animaux_proteine + df_alimentation_vegetal.loc[:,'Disponibilite alimentaire proteine (kg/an)'] 
total_disponibilite_sans_perte_vegetal_proteine = df_alimentation_vegetal['Disponibilite sans perte proteine(kg/an)'].sum() * 1000
print("total_disponibilite_sans_perte_vegetal_proteine en g= ",total_disponibilite_sans_perte_vegetal_proteine)

total_disponibilite_sans_perte_vegetal_proteine en g=  167942868574310.88


total_disponibilite_sans_perte_vegetal_proteine en g=  167942868574310.88


In [64]:
nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine =total_disponibilite_sans_perte_vegetal_proteine / (49 * 365)
pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine = nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine *100/nb_population
print ('nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine =', nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine)
print('pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine =', pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine,'%')

nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine = 9390152003.036673
pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine = 134.1962915982001 %


nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine = 9390152003.036673  <br>
pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine = 134.1962915982001 %

# Question 9 : Combien d'humains pourraient être nourris avec la disponibilité alimentaire mondiale ? Donnez les résultats en termes de calories, puis de protéines, et exprimez ensuite ces 2 résultats en pourcentage de la population mondiale.

In [65]:
# Vegetal ou pas ?

<b>a. Selon le nombre de kcal :</b>

In [66]:
df_prinicpale.columns

Index(['Zone', 'Code zone', 'Produit', 'Aliments pour animaux (kg/an)',
       'Autres utilisations (non alimentaire) (kg/an)',
       'Disponibilité alimentaire (Kcal/personne/jour)',
       'Disponibilité alimentaire en quantité (kg/personne/an)',
       'Disponibilité de matière grasse en quantité (g/personne/jour)',
       'Disponibilité de protéines en quantité (g/personne/jour)',
       'Disponibilité intérieure (kg/an)', 'Exportations - Quantité (kg/an)',
       'Importations - Quantité (kg/an)', 'Nourriture (kg/an)',
       'Pertes (kg/an)', 'Production (kg/an)', 'Semences (kg/an)',
       'Traitement (kg/an)', 'Variation de stock (kg/an)', 'Type',
       'Population', 'Unité', 'Disponibilite alimentaire (kcal/an)',
       'Disponibilite alimentaire proteine (kg/an)',
       'Ratio Energie/poids (kcal/kg)', 'Ratio protéine/poids'],
      dtype='object')

In [67]:
disponibilite_globale_kcal = df_prinicpale['Disponibilite alimentaire (kcal/an)'].sum()

In [68]:
nb_personne_nourris_disponibilite_gobale_kcal =disponibilite_globale_kcal / (2500 * 365)
pourcentage_personne_nourris_disponibilite_gobale_kcal = nb_personne_nourris_disponibilite_gobale_kcal *100 / nb_population
print('disponibilite_globale_kcal = ',disponibilite_globale_kcal)
print ('nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal =', nb_personne_nourris_disponibilite_gobale_kcal)
print('pourcentage_nourris_disponibilite_sans_perte_vegetal_kcal =', pourcentage_personne_nourris_disponibilite_gobale_kcal,'%')

disponibilite_globale_kcal =  7364437353675000.0
nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal = 8070616278.0
pourcentage_nourris_disponibilite_sans_perte_vegetal_kcal = 115.33857759378368 %


disponibilite_globale_kcal =  7364437353675000.0  <br>
nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal = 8070616278.0 <br>
pourcentage_nourris_disponibilite_sans_perte_vegetal_kcal = 115.33857759378368 % <br>

<b>b. Selon le nombre de proteines : </b>

In [69]:
disponibilite_globale_proteine = df_prinicpale['Disponibilite alimentaire proteine (kg/an)'].sum() * 1000

In [70]:
nb_personne_nourris_disponibilite_gobale_proteine =disponibilite_globale_proteine / (49 * 365)
pourcentage_personne_nourris_disponibilite_gobale_proteine = nb_personne_nourris_disponibilite_gobale_proteine *100 / nb_population
print('disponibilite_globale_proteine = ',disponibilite_globale_proteine)
print ('nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine =', nb_personne_nourris_disponibilite_gobale_proteine)
print('pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine =', pourcentage_personne_nourris_disponibilite_gobale_proteine,'%')

disponibilite_globale_proteine =  207249144517349.97
nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine = 11587875007.959183
pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine = 165.60433239724978 %


disponibilite_globale_proteine =  207249144517349.97  <br>
nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine = 11587875007.959183 <br>
pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine = 165.60433239724978 % <br>


# Question 10 : A partir des données téléchargées qui concernent la sous-nutrition, répondez à cette question : Quelle proportion de la population mondiale est considérée comme étant en sous-nutrition ?


In [71]:
df_sous_alimentation20132015 = df_sous_alimentation[df_sous_alimentation['Code année']==20132015].copy()

In [72]:
# df_sous_alimentation2013[df_sous_alimentation2013['Symbole']!='F']

In [73]:
# df_sous_alimentation20132015['Valeur'].fillna(0)

In [74]:
# df['a'] = pd.to_numeric(df['a'],errors = 'coerce')
df_sous_alimentation20132015.loc[:,'Valeur'] = pd.to_numeric(df_sous_alimentation20132015.loc[:,'Valeur'], errors = "coerce")

In [75]:
nb_sous_alimentation20132015 = df_sous_alimentation20132015['Valeur'].sum() * 1000000
print('nb_sous_alimentation20132015 = ',nb_sous_alimentation20132015)
pourcentage_sous_alimentation20132015  = nb_sous_alimentation20132015 *100/ nb_population
print('pourcentage_sous_alimentation20132015 = ',pourcentage_sous_alimentation20132015)

nb_sous_alimentation20132015 =  735200000.0
pourcentage_sous_alimentation20132015 =  10.506870767490325


nb_sous_alimentation20132015 =  735200000.0 <br>
pourcentage_sous_alimentation20132015 =  10.506870767490325

# Question 11 : En ne prenant en compte que les céréales destinées à l'alimentation (humaine et animale), quelle proportion (en termes de poids) est destinée à l'alimentation animale ?

In [76]:
# df_cereales["Produit"].apply(lambda d = d.drop_duplicates())

In [77]:
# df.apply(lambda x: '-'.join(x), axis = 1)

In [78]:
# df_cereales.groupby('Produit').apply(lambda d: d.drop_duplicates())
df_cereales.head(3)

Code Domaine                                            Domaine  Code zone  \
0         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
1         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
2         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   

          Zone  Code Élément     Élément  Code Produit           Produit  \
0  Afghanistan          5511  Production          2511               Blé   
1  Afghanistan          5511  Production          2805  Riz (Eq Blanchi)   
2  Afghanistan          5511  Production          2513              Orge   

   Code année  Année               Unité  Valeur Symbole  \
0        2013   2013  Milliers de tonnes    5169       S   
1        2013   2013  Milliers de tonnes     342       S   
2        2013   2013  Milliers de tonnes     514       S   

  Description du Symbole  
0  Données standardisées  
1  Données standardisées  
2  Données standardisées

<b> a. Liste des produits considérés comme des céréales selon la FAO.</b>

In [79]:
df_type_cereal = df_cereales.drop_duplicates(subset = ['Code Produit', 'Produit'])
df_type_cereal[['Code Produit', 'Produit']]

Code Produit           Produit
0           2511               Blé
1           2805  Riz (Eq Blanchi)
2           2513              Orge
3           2514              Maïs
4           2517            Millet
9           2515            Seigle
10          2516            Avoine
12          2518            Sorgho
13          2520  Céréales, Autres

<b> b. Rajout de la colonne "is_cereal"de type booléen pour distinguer les céréales des autres produits </b>

In [80]:
type_cereal = df_type_cereal["Produit"].tolist()
df_prinicpale['is_cereal'] = df_prinicpale['Produit'].apply(lambda x:True if x in type_cereal else False)

In [117]:
df_prinicpale_cereal = df_prinicpale[df_prinicpale['is_cereal']==True].copy()
# df_prinicpale_cereal = df_prinicpale_cereal[['Zone','Produit','Aliments pour animaux (kg/an)', 'Nourriture (kg/an)']].copy()

In [118]:
df_prinicpale_cereal[['Zone','Produit','Aliments pour animaux (kg/an)', 'Nourriture (kg/an)']]

Zone Produit  Aliments pour animaux (kg/an)  \
4469             Afghanistan     Blé                            NaN   
4470          Afrique du Sud     Blé                   3.700000e+07   
4471                 Albanie     Blé                   1.800000e+07   
4472                 Algérie     Blé                   5.450000e+08   
4473               Allemagne     Blé                   7.494000e+09   
...                      ...     ...                            ...   
15271    Émirats arabes unis  Sorgho                   5.900000e+07   
15272               Équateur  Sorgho                   1.400000e+07   
15273  États-Unis d'Amérique  Sorgho                   2.351000e+09   
15274               Éthiopie  Sorgho                            NaN   
15275           Îles Salomon  Sorgho                   0.000000e+00   

       Nourriture (kg/an)  
4469         4.895000e+09  
4470         3.174000e+09  
4471         4.400000e+08  
4472         7.270000e+09  
4473         6.900000e+09  
...                   ...  
15271                 NaN  
15272                 NaN  
15273        2.360000e+08  
15274        2.401000e+09  
15275                 NaN  

[1470 rows x 4 columns]

In [119]:
cereales_animaux = df_prinicpale_cereal['Aliments pour animaux (kg/an)'].sum()
cereales_nourriture = df_prinicpale_cereal['Nourriture (kg/an)'].sum()

proportion_cereales_animaux = 100 * cereales_animaux / (cereales_animaux + cereales_nourriture)
#proportion_cereales_nourriture = 100 * cereales_nourriture / (cereales_animaux + cereales_nourriture)

print('proportion_cereales_animaux = ', proportion_cereales_animaux)
#print('proportion_cereales_nourriture = ',proportion_cereales_nourriture)


proportion_cereales_animaux =  45.91400584899829



# Repérez les 15 produits les plus exportés par ce groupe de pays.

Parmi les données des bilans alimentaires au niveau mondial, sélectionnez les 200 plus grandes importations de ces produits (1 importation = une quantité d'un produit donné importée par un pays donné)

Groupez ces importations par produit, afin d'avoir une table contenant 1 ligne pour chacun des 15 produits. Ensuite, calculez pour chaque produit les 2 quantités suivantes :

 

le ratio entre la quantité destinés aux "Autres utilisations" (Other uses) et la disponibilité intérieure.
le ratio entre la quantité destinée à la nourriture animale et la quantité destinée à la nourriture (animale + humaine)
 # Question 12 : Donnez les 3 produits qui ont la plus grande valeur pour chacun des 2 ratios (vous aurez donc 6 produits à citer).

Sélectionnez parmi les données des bilans alimentaires les informations relatives aux pays dans lesquels la FAO recense des personnes en sous-nutrition.


<b> Les pays en sous alimentation</b>

In [120]:
df_pays_sous_alimentation = df_sous_alimentation20132015[~df_sous_alimentation20132015['Valeur'].isin([0, np.NaN])]
# df_pays_sous_alimentation[['Code zone','Zone']]

In [121]:
liste_code_pays_sous_alimentation = df_pays_sous_alimentation['Code zone'].tolist()
df_prinicpale['is_malnutrition'] = df_prinicpale['Code zone'].apply(lambda y: True if y in liste_code_pays_sous_alimentation else False)

<b> Rajout de la colonne 'is_malnutrition' dans la data frame principale

In [122]:
df_prinicpale_sous_alimentation = df_prinicpale[df_prinicpale['is_malnutrition']==True]


In [123]:
# df_prinicpale_sous_alimentation.columns

<b>les 15 produits les plus exportés par ce groupe de pays.

In [124]:
produit_exporte_par_pays_sous_alim = df_prinicpale_sous_alimentation.groupby("Produit").sum().reset_index()[['Produit','Exportations - Quantité (kg/an)']].sort_values(by='Exportations - Quantité (kg/an)', ascending=False)
liste_produit_exporte_par_pays_sous_alim = produit_exporte_par_pays_sous_alim.head(15)['Produit'].to_list()
liste_produit_exporte_par_pays_sous_alim


['Huile de Palme',
 'Maïs',
 'Manioc',
 'Riz (Eq Blanchi)',
 'Sucre Eq Brut',
 'Blé',
 'Légumes, Autres',
 'Bananes',
 'Fruits, Autres',
 'Lait - Excl Beurre',
 'Soja',
 'Poissons Pelagiques',
 'Tomates',
 'Pommes',
 'Oranges, Mandarines']

<b>Parmi les données des bilans alimentaires au niveau mondial, sélectionnez les 200 plus grandes importations de ces produits (1 importation = une quantité d'un produit donné importée par un pays donné)

In [125]:
# df_prinicpale_sous_alimentation.groupby('Produit')['Exportations - Quantité (kg/an)'].rank(method='max') == 2 

In [126]:
df_prinicpale_sous_alimentation_produit_exporte = df_prinicpale_sous_alimentation[df_prinicpale_sous_alimentation['Produit'].isin(liste_produit_exporte_par_pays_sous_alim)].sort_values(by='Exportations - Quantité (kg/an)', ascending=False).head(200).copy()
# df_prinicpale_sous_alimentation_produit_exporte = df_prinicpale_sous_alimentation_produit_exporte[['Zone','Code zone','Produit','Exportations - Quantité (kg/an)']].copy()
#df_prinicpale_cereal = df_prinicpale_cereal[['Zone','Produit','Aliments pour animaux (kg/an)', 'Nourriture (kg/an)']].copy()

In [127]:
# liste_sous_alimentation_produit_exporte = df_prinicpale_sous_alimentation_produit_exporte['Produit'].head(200)

In [128]:
# df_prinicpale_sous_alimentation_produit_exporte

In [129]:
df_prinicpale_sous_alimentation_produit_exporte

Zone  Code zone           Produit  \
13722                        Thaïlande        216            Manioc   
7142                         Indonésie        101    Huile de Palme   
8070                         Argentine          9              Maïs   
7164                          Malaisie        131    Huile de Palme   
10297                             Inde        100  Riz (Eq Blanchi)   
...                                ...        ...               ...   
10422  Bolivie (État plurinational de)         19     Sucre Eq Brut   
10567                         Zimbabwe        181     Sucre Eq Brut   
4530                           Géorgie         73               Blé   
9833                       Ouzbékistan        235            Pommes   
5901                          Malaisie        131    Fruits, Autres   

       Aliments pour animaux (kg/an)  \
13722                   1.800000e+09   
7142                             NaN   
8070                    4.923000e+09   
7164                             NaN   
10297                   2.124000e+09   
...                              ...   
10422                            NaN   
10567                            NaN   
4530                    6.700000e+07   
9833                    1.270000e+08   
5901                             NaN   

       Autres utilisations (non alimentaire) (kg/an)  \
13722                                   2.081000e+09   
7142                                    2.514000e+09   
8070                                    1.266000e+09   
7164                                    3.930000e+08   
10297                                            NaN   
...                                              ...   
10422                                   6.000000e+07   
10567                                   0.000000e+00   
4530                                    3.000000e+06   
9833                                             NaN   
5901                                    0.000000e+00   

       Disponibilité alimentaire (Kcal/personne/jour)  \
13722                                            40.0   
7142                                            116.0   
8070                                             86.0   
7164                                            164.0   
10297                                           689.0   
...                                               ...   
10422                                           264.0   
10567                                           261.0   
4530                                           1200.0   
9833                                             29.0   
5901                                             21.0   

       Disponibilité alimentaire en quantité (kg/personne/an)  \
13722                                              13.00        
7142                                                4.80        
8070                                               10.44        
7164                                                6.89        
10297                                              69.49        
...                                                  ...        
10422                                              26.60        
10567                                              28.70        
4530                                              163.43        
9833                                               22.11        
5901                                               16.78        

       Disponibilité de matière grasse en quantité (g/personne/jour)  \
13722                                               0.05               
7142                                               13.16               
8070                                                0.27               
7164                                               18.54               
10297                                               1.53               
...                                                  ...               
10422                                                NaN 

<b> Groupez ces importations par produit, afin d'avoir une table contenant 1 ligne pour chacun des 15 produits. Ensuite, calculez pour chaque produit les 2 quantités suivantes :

In [130]:
gp_prinicpale_sous_alimentation_produit_exporte = df_prinicpale_sous_alimentation_produit_exporte.groupby(['Produit']).sum().reset_index().copy()

<b>le ratio entre la quantité destinés aux "Autres utilisations" (Other uses) et la disponibilité intérieure.</b><br>
<b>le ratio entre la quantité destinée à la nourriture animale et la quantité destinée à la nourriture (animale + humaine)</b>

In [131]:
gp_prinicpale_sous_alimentation_produit_exporte['Ratio autres utilisations/ Disponibilité intèrieur'] = gp_prinicpale_sous_alimentation_produit_exporte['Autres utilisations (non alimentaire) (kg/an)'] / gp_prinicpale_sous_alimentation_produit_exporte['Disponibilité intérieure (kg/an)']


In [132]:
gp_prinicpale_sous_alimentation_produit_exporte['Ratio nourriture animale/nourriture'] = \
gp_prinicpale_sous_alimentation_produit_exporte['Aliments pour animaux (kg/an)'] / \
gp_prinicpale_sous_alimentation_produit_exporte['Nourriture (kg/an)'] + gp_prinicpale_sous_alimentation_produit_exporte['Aliments pour animaux (kg/an)']

In [168]:
gp_prinicpale_sous_alimentation_produit_exporte.head()

Produit  Code zone  Aliments pour animaux (kg/an)  \
0             Bananes       1220                   8.070000e+08   
1                 Blé       3386                   3.932400e+10   
2      Fruits, Autres       3052                   0.000000e+00   
3      Huile de Palme       1114                   0.000000e+00   
4  Lait - Excl Beurre       2754                   3.079000e+10   

   Autres utilisations (non alimentaire) (kg/an)  \
0                                   9.500000e+07   
1                                   6.505000e+09   
2                                   5.300000e+07   
3                                   5.056000e+09   
4                                   1.043000e+09   

   Disponibilité alimentaire (Kcal/personne/jour)  \
0                                           435.0   
1                                         15852.0   
2                                           652.0   
3                                          1373.0   
4                                          3032.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                             279.29        
1                                            2018.45        
2                                             535.24        
3                                              56.95        
4                                            1938.43        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               1.22               
1                                              78.23               
2                                              10.45               
3                                             155.31               
4                                             164.83               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               5.92          
1                                             463.23          
2                                               8.02          
3                                               0.06          
4                                             176.63          

   Disponibilité intérieure (kg/an)  Exportations - Quantité (kg/an)  ...  \
0                      1.351300e+10                     1.706500e+10  ...   
1                      3.209140e+11                     2.294800e+10  ...   
2                      1.212740e+11                     1.384300e+10  ...   
3                      7.467000e+09                     4.539500e+10  ...   
4                      2.768080e+11                     1.333100e+10  ...   

   Variation de stock (kg/an)  Population  \
0                4.800000e+08      437070   
1               -2.196000e+09     3723770   
2                3.980000e+08     3581318   
3               -1.220000e+08      468751   
4                2.460000e+08     3437163   

   Disponibilite alimentaire (kcal/an)  \
0                         6.430704e+12   
1                         7.569173e+14   
2                         4.479602e+13   
3                         2.095845e+13   
4                         1.365492e+14   

   Disponibilite alimentaire proteine (kg/an)  Ratio Energie/poids (kcal/kg)  \
0                                8.649056e+07                    7325.371956   
1                                2.267682e+10                   62645.509490   
2                                5.133201e+08                    8306.580340   
3                                4.863990e+05                   87194.975382   
4                                8.136873e+09                   11147.938293   

   Ratio protéine/poids  is_cereal  is_malnutrition  \
0              0.098796          0               12   
1              1.802940         22               22   
2              0.102097          0               19   
3              0.049266          0               10   
4              0.638273          0               19   

   Ratio autres utilisati

<b>Donnez les 3 produits qui ont la plus grande valeur pour chacun des 2 ratios (vous aurez donc 6 produits à citer).</b>

In [133]:
liste_ratio1 = gp_prinicpale_sous_alimentation_produit_exporte.sort_values(by='Ratio autres utilisations/ Disponibilité intèrieur',ascending=False).head(3)['Produit'].to_list()

In [134]:
liste_ratio2 = gp_prinicpale_sous_alimentation_produit_exporte.sort_values(by='Ratio nourriture animale/nourriture',ascending=False).head(3)['Produit'].to_list()

In [167]:
liste_ratio = liste_ratio1 + liste_ratio2
print('liste_ratio1 = ',liste_ratio1)
print('liste_ratio2 = ',liste_ratio2)
print('liste_ratio = ',liste_ratio)

liste_ratio1 =  ['Huile de Palme', 'Manioc', 'Maïs']
liste_ratio2 =  ['Maïs', 'Blé', 'Légumes, Autres']
liste_ratio =  ['Huile de Palme', 'Manioc', 'Maïs', 'Maïs', 'Blé', 'Légumes, Autres']


liste_ratio1 =  ['Huile de Palme', 'Manioc', 'Maïs']
liste_ratio2 =  ['Maïs', 'Blé', 'Légumes, Autres']
liste_ratio =  ['Huile de Palme', 'Manioc', 'Maïs', 'Maïs', 'Blé', 'Légumes, Autres']

# Question 13 : Combien de tonnes de céréales pourraient être libérées si les USA diminuaient leur production de produits animaux de 10% ?

In [158]:
df_cereal_USA = df_prinicpale_cereal[df_prinicpale_cereal['Zone']=="États-Unis d'Amérique"].groupby('Zone').sum().copy()

In [166]:
cereal_libere_usa = (df_cereal_USA['Aliments pour animaux (kg/an)'].iloc[0]*10 /100 ) /1000
print('cereal_libere_usa = ',cereal_libere_usa)

cereal_libere_usa =  14009600.0


cereal_libere_usa =  14009600.0

# Question 14 : En Thaïlande, quelle proportion de manioc produit est exportée ? Quelle est la proportion de personnes en sous-nutrition ?